File: Cores/ExtremumScalaCore/src/main/scala/OSOL/Extremum/Core/Scala/Optimization/Nodes/SetParametersNode.scala
package OSOL.Extremum.Core.Scala.Optimization.Nodes

import OSOL.Extremum.Core.Scala.Optimization._
import OSOL.Extremum.Core.Scala.Optimization.Exceptions._

class SetParametersNode[Base, FuncType, V <: Optimizable[Base, FuncType]]
(override val nodeId: Int, val parameters: Map[String, Any])
  extends GeneralNode[Base, FuncType, V ](nodeId) {

  final override def initialize(f: Map[String, FuncType] => FuncType, area: Area, state: State[Base, FuncType, V]): Unit = {
    parameters.foreach { case (k, v) => state.setParameter(k, v)}
  }

  final override def process(f: Map[String, FuncType] => FuncType, area: Area, state: State[Base, FuncType, V]): Unit = { }

}

File: Cores/OSOL.Extremum.Core.DotNet/Optimization/Nodes/SetParametersNode.cs
using System;
using System.Collections.Generic;

namespace OSOL.Extremum.Core.DotNet.Optimization.Nodes
{
    
    using Area = Dictionary<string, Tuple<double, double>>;

    public class SetParametersNode<TBase, TFuncType, TV>: GeneralNode<TBase, TFuncType, TV>
        where TV: class, IOptimizable<TBase, TFuncType>
    {
        public Dictionary<string, object> Parameters;
        
        public SetParametersNode(int nodeId, Dictionary<string, object> parameters)
        {
            this.NodeId = nodeId;
            this.Parameters = parameters;
        }
        
        public override void Initialize(Func<Dictionary<string, TFuncType>, TFuncType> f, Area area, State<TBase, TFuncType, TV> state)
        {
            foreach (var kvp in Parameters)
            {
                state.SetParameter(kvp.Key, kvp.Value);
            }
        }

        public override void Process(Func<Dictionary<string, TFuncType>, TFuncType> f, Area area, State<TBase, TFuncType, TV> state)
        {
            // No action required
        }
    }
}